src/transactions/TransactionFrame.cpp

```diff
@@ -300,28 +300,28 @@ TransactionFrame::makeOperation(Operation const& op, OperationResult& res,
 
 void
 TransactionFrame::resetResults(LedgerHeader const& header, int64_t baseFee,
```

Comment:
Seems like this method still mutates `TransactionFrame` (it has `mOperations.clear()` call). Not sure if that's in the scope of this PR, just wanted to make sure this doesn't escape our attention.

Reply:
Yes I also noticed that, I agree this is one place that needs attention.

Reply:
this is because the change in this PR is incomplete: if you look at `OperationFrame`, it has a `mResult` that gets updated in a bunch of places.
Like when "apply" is called -> it ends up calling `OperationFrame::doApply` that mutates the result with things like `innerResult().code(CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);`
I think it's just that `doApply` is missing a parameter so that it uses the result passed to `TransactionFrame::apply` (same thing in other places like `doCheckValid`

Reply:
After you remove `mResult from `OperationFrame` `mOperations` should be set once and be immutable

Reply:
the reason this is incomplete and unsafe is that `OperationFrame::mResult` ends up _caching_ a reference to the Result that you use, this in turn can have catastrophic implications in production code if the the local variable goes out of scope before the TransactionFrame gets deleted.